commons/src/main/java/com/powsybl/commons/config/MapModuleConfig.java

```diff
@@ -126,74 +154,118 @@ public void setStringListProperty(String name, List<String> stringList) {
         return strings.stream().map(s -> Enum.valueOf(clazz, s)).collect(Collectors.toSet());
     }
 
+    @Override
+    public Optional<Integer> getOptionalIntegerProperty(String name) {
+        Objects.requireNonNull(name);
+        Object value = properties.get(name);
+        if (value == null) {
+            return Optional.empty();
+        }
+        if (value instanceof Integer) {
+            return Optional.of((Integer) value);
+        } else if (value instanceof String) {
+            return Optional.of(Integer.parseInt((String) value));
```

Comment:
It could be interesting to catch the exception if the string value is not an integer and to rethrow an unexpectedTypeException to help user to understand the error.

Reply:
Done